rust/src/delta.rs

```diff
@@ -1463,9 +1415,7 @@ impl<'a> DeltaTransaction<'a> {
                 }
                 Err(e) => {
                     match e {
-                        DeltaTableError::TransactionError {
-                            source: DeltaTransactionError::VersionAlreadyExists { .. },
-                        } => {
+                        DeltaTableError::VersionAlreadyExists(_) => {
```

Comment:
Also I like this better. 
btw @houqp, do we have a reason why we use struct like errors instead of value based? I remember something regarding docs but cannot find what exactly

Reply:
What do you mean by value based errors?

Reply:
e.g. A vs B in an example:
```rust
enum Error {
  A {
    source: String
  },
  B(String),
}
```

Reply:
Oh i see. it's just for thiserror crate so it can implement the from trait automatically for us. See discussion related to From trait in https://docs.rs/thiserror/1.0.29/thiserror/index.html. It's mostly only useful for wrapping error type values though. For types like String, I don't think there is much value in storing it in a source field.